src/Rcl/SillyArrayMap.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE TypeApplications #-}
+
+-- | Stackage changed its JSON representation for package differences
+--
+-- It used to be nested objects:
+--
+-- @
+-- { diff:
+--   { some-package:    { lts-x: version-x, lts-y: version-y }
+--   , another-package: { lts-x: version-x, lts-y: version-y }
+--   , ...
+--   }
+-- }
+-- @
+--
+-- which was very easy to work with in Haskell (map of maps).
+--
+-- Now it's this:
+--
+-- @
+-- { diff:
+--   [ ["some-package",    { lts-x: version-x, lts-y: version-y }]
+--   , ["another-package", { lts-x: version-x, lts-y: version-y }]
+--   , ...
+--   ]
+-- }
+-- @
+--
+-- Which is /not/ easy to work with in Haskell (list of heterogenious list).
+--
+-- I guess is nicer for JavaScript? Any, I'm representing this silly type and
```

Comment:
I dunno why they did this, this isn't any easier to use in JavaScript. At first I thought maybe they were using the `TwoElemArray` constructor from [`SumEncoding`](https://www.stackage.org/haddock/lts-15.10/aeson-1.4.7.1/Data-Aeson.html#t:SumEncoding), but since the key is an arbitrary package, it's more like an open sum. V strange